### PR TITLE
Adding accordion JS and style updates.

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Adding JS to the accordion and contextual menu

## QA

- pull branch and view the accordion and contextual menu pages
- Review code and test functionality

### CODE REVIEW

File: examples/patterns/accordion/accordion.html [hunk 1]
```diff
@@ -7,22 +7,139 @@
...
+     *
+     * @param {String} tabCtrl
+     *
+     * @example
+     * <button class="p-accordion__tab" id="status-tab" role="tab" aria-controls="#status" aria-expanded="false" aria-selected="false">Status</button>
+     * <section class="p-accordion__panel" id="status" role="tabpanel" aria-hidden="true" aria-labelledby="status-tab">
+     *     <p>Closed panel</p>
+     * </section>
+     */
+
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: no space please

File: examples/patterns/accordion/accordion.html [hunk 2]
```diff
@@ -7,22 +7,139 @@
...
+    function accordionToggles(tabCtrl) {
+
+        // Collapse fucntion
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Spelling mistake

File: examples/patterns/accordion/accordion.html [hunk 3]
```diff
@@ -7,22 +7,139 @@
...
+        function collapse (tab) {
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Add some docstring

File: examples/patterns/accordion/accordion.html [hunk 4]
```diff
@@ -7,22 +7,139 @@
...
+            var target = tab.getAttribute('aria-controls');
+            var panel = document.querySelector(target);
+
+            panel.setAttribute('aria-hidden', true);
+            tab.setAttribute('aria-expanded', false);
+        }
+
+        // Open function
+        function open (tab) {
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Add docstring

Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Very similar functionality to collapse, perhaps combine in to one function that toggles/ accepts a param?

File: examples/patterns/accordion/accordion.html [hunk 5]
```diff
@@ -7,22 +7,139 @@
...
+            var target = tab.getAttribute('aria-controls');
+            var panel = document.querySelector(target);
+
+            panel.setAttribute('aria-hidden', false);
+            tab.setAttribute('aria-expanded', true);
+        }
+
+        // Collapse all tabs and content
+        function collapseAll (event) {
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: docstring

File: examples/patterns/contextual-menu/contextual-menu.html [hunk 6]
```diff
@@ -68,3 +68,99 @@
...
+     * </div>
+     *
+     */
+
+    function toggle(toggleCtrl) {
+
+        // Select all toggle control elements
+        var t = document.querySelectorAll(toggleCtrl);
+
+        console.log(t);
```
Comment (Lukewh) [inline examples/patterns/contextual-menu/contextual-menu.html]: Remove logging

File: examples/patterns/contextual-menu/contextual-menu.html [hunk 7]
```diff
@@ -68,3 +68,99 @@
...
+     * @param {String} toggleCtrl
+     *
+     * @example
+     * <button class="u-toggle" aria-controls="#menu" aria-expanded="false">Toggle</button>
+     * <div id="menu" aria-hidden="true">
+     *     <p>Example menu</p>
+     * </div>
+     *
+     */
+
```
Comment (Lukewh) [inline examples/patterns/contextual-menu/contextual-menu.html]: No space.

**Comment by richmccartney** [comment]
@Lukewh Everyhting should be fixed now if you take a look

File: examples/patterns/accordion/accordion.html [hunk 8]
```diff
@@ -7,22 +7,157 @@
...
+
+        // Select all toggle control elements
+        var toggles = Array.prototype.slice.call(
+          document.querySelectorAll(tabCtrl)
+        );
+
+        /**
+         * Set click event to each toggle control element
+         * @param {String} toggle
+         */
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Don't need docstring for this

File: examples/patterns/accordion/accordion.html [hunk 9]
```diff
@@ -7,22 +7,157 @@
...
+              toggle.setAttribute('aria-expanded', false);
+          }
+        }
+
+        /**
+         * Toggle all function
+         * @param {String} target
+         * @param {String} toggles
+         */
+        function toggleAll (event) {
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Docstring wrong - only param is the event

File: examples/patterns/accordion/accordion.html [hunk 10]
```diff
@@ -7,22 +7,157 @@
...
+     *     <p>Closed panel</p>
+     * </section>
+     */
+    function accordionToggles(tabCtrl) {
+
+        /**
+         * Toggle function
+         * @param {String} toggle
+         */
+        function toggle (toggle, open) {
```
Comment (Lukewh) [inline examples/patterns/accordion/accordion.html]: Docstring missing `open` param

**Comment by richmccartney** [comment]
@Lukewh cna you take one further look please

**Comment by Lukewh** [comment]
## QA:
Context menu - should clicking anywhere off the menu close it? If so - uh-oh, if not - no worries.

**Comment by richmccartney** [comment]
@Lukewh it should.. I'll make a separate PR

**Pull request merged**